File: java/src/shortener/Base62.java
package shortener;

public class Base62 {
    public static final String ALPHABET = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";

    public static final int BASE = ALPHABET.length();

    public static String encode(int i) {
        StringBuilder sb = new StringBuilder("");
        if (i == 0) {
            return "a";
        }
        while (i > 0) {
            int rem = i % BASE;
            sb.append(ALPHABET.charAt(rem));
            i = i / BASE;
        }
        return sb.reverse().toString();
    }
}

File: kotlin/src/shortener/Base62.kt
package shortener

object Base62 {
    private val ALPHABET = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
    private val BASE = ALPHABET.length

    fun encode(num: Int): String {
        var i = num
        val sb = StringBuilder("")
        if (i == 0) {
            return "a"
        }
        while (i > 0) {
            val rem = i % BASE
            sb.append(ALPHABET[rem])
            i /= BASE
        }
        return sb.reverse().toString()
    }
}
